### ISSUE TITLE
Allow @JsonIgnore processor to throw error instead of silently ignoring the annotated property

### ISSUE THREAD

**Issue by joca-bt** [FasterXML/jackson-databind#3850]
We have a situation in which we have ignored properties that when provided by the user we would like to throw an error about the property not existing instead of silently ignoring them.

`@JsonIgnore` documentation states:

> When ignoring the whole property, the default behavior if encountering such property in input is to ignore it without exception; but if there is a [JsonAnySetter](https://javadoc.io/static/com.fasterxml.jackson.core/jackson-annotations/2.14.2/com/fasterxml/jackson/annotation/JsonAnySetter.html) it will be called instead. Either way, no exception will be thrown.

Is this feature something that could make sense? Maybe there is already a way of achieving this with standard Jackson, can you share if so?

**Comment by mukham12** [comment]
Are you looking for something like this @joca-bt :

`ObjectMapper mapper = new ObjectMapper();
mapper.configure(DeserializationFeature.FAIL_ON_IGNORED_PROPERTIES, true);
`

**Comment by cowtowncoder** [comment]
@joca-bt Yes, but to me it sounds like this setting does exactly what you ask -- fails (throws exception) on properties marked as ignored, if they are encountered in content.

**Comment by joca-bt** [comment]
I will try to upload a test case later, as errors are not being thrown.

**Comment by joca-bt** [comment]
The feature works, thanks! I misread the original comment.
